pkg/scalers/aws_cloudwatch_scaler.go

```diff
@@ -139,26 +146,35 @@ func parseAwsCloudwatchMetadata(metadata, resolvedEnv map[string]string) (*awsCl
 		return nil, fmt.Errorf("no awsRegion given")
 	}
 
-	accessIDKey := awsAccessKeyIDEnvVar
-	if val, ok := metadata["awsAccessKeyID"]; ok && val != "" {
-		accessIDKey = val
-	}
-
-	if val, ok := resolvedEnv[accessIDKey]; ok && val != "" {
-		meta.awsAccessKeyID = val
+	// Check TriggerAuthentication first
+	if podIdentity == "aws-credentials" {
+		meta.awsAccessKeyID = authParams["awsAccessKeyId"]
+		meta.awsSecretAccessKey = authParams["awsSecretAccessKey"]
+	} else if podIdentity == "aws-role" {
+		meta.awsRoleArn = authParams["awsRoleArn"]
 	} else {
-		return nil, fmt.Errorf("cannot find awsAccessKeyId named %s in pod environment", accessIDKey)
-	}
+		// then fallback to what is defined on the pod
```

Comment:
Auth information should no longer be defined in the metadata, only in the trigger auth